src/backend/distributed/utils/metadata_cache.c

```diff
@@ -614,6 +624,17 @@ DistPartitionLogicalRelidIndexId(void)
 }
 
 
+/* return oid of pg_dist_partition_colocationid_index index */
+Oid
+DistPartitionColocationidIndexId(void)
```

Comment:
You need to ensure that these are invalidated once the cache invalidated. See `medata_cache.c@1029`
We should also invalidate relationId itself.